src/test/kotlin/ecommerce/controller/ProductControllerTest.kt

```diff
@@ -0,0 +1,110 @@
+package ecommerce.controller
+
+import ecommerce.model.Product
+import io.restassured.RestAssured
+import io.restassured.http.ContentType
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpStatus
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.jdbc.support.GeneratedKeyHolder
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+class ProductControllerTest {
+    @Autowired
+    private lateinit var jdbcTemplate: JdbcTemplate
```

Comment:
Since ProductController depends on ProductRepository, it’s better to bring in ProductRepository and test it!